.github/workflows/deploy.yml

```diff
@@ -48,42 +56,72 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # https://github.com/actions/checkout/releases/tag/v3.5.3
-      - name: Set preview in package.json
+
+      - name: Set prerelease in package.json
+        if: inputs.deploy_type == 'prerelease'
         id: set-preview
         run: |
           ./build/preview.sh
         env:
           LANGUAGE_SERVER_VERSION: ${{ github.event.inputs.langserver }}
+
       - name: Setup Node
         uses: actions/setup-node@e33196f7422957bea03ed53f6fbb155025ffc7b8 # https://github.com/actions/setup-node/releases/tag/v3.7.0
         with:
           node-version-file: '.nvmrc'
+
       - name: Install dependencies
         run: npm ci
         env:
           npm_config_arch: ${{ matrix.npm_config_arch }}
           ls_target: ${{ matrix.ls_target }}
-      - name: Package VSIX
-        run: npm run package -- --pre-release --target=${{ matrix.vsce_target }}
+
+      - name: Package PreRelease VSIX
+        if: inputs.deploy_type == 'prerelease'
+        run: npm run package -- --githubBranch $GITHUB_REF_NAME --pre-release --target=${{ matrix.vsce_target }}
+
+      - name: Package Stable VSIX
+        if: inputs.deploy_type == 'stable'
+        run: npm run package -- --target=${{ matrix.vsce_target }}
```

Comment:
I still don't entirely understand why this needs to be separated and why we couldn't just use the `githubBranch` flag in either case, but I suppose it's not a blocker.

Reply:
Line 81 also contains a required `--pre-release` flag

Reply:
Oh, I missed that one! Good catch and apologies for the confusion!

Reply:
I still think we could be passing `--githubBranch $GITHUB_REF_NAME` for stable releases as well but again - it's not a blocker.